EVES/drafts/EVES-005/eves-005.md

```diff
@@ -76,10 +76,26 @@ The consumer uses metadata search or similar services to identify an asset of in
 
 At this point, the contract is fully settled and the consumer can use the asset.
 
-### 5. Fee Payment
+### 5. Fee and Payment Report
 
-On a regular basis, the providers send the accumulated fees to the operator.
-To ensure the fees are correctly calculated without publishing business statistics, a provider should construct a zero knowledge proof for the accumulated amount being correct based on the submitted hashes.
+#### 5.1. Asset Fee
+
+1. After a defined period, the provider compiles all completed contracts and generates a cumulative bill.
+2. The provider sends this bill to the respective customers.
+3. The customers review the bill and make payments.
+
+#### 5.2. Operator Fee
+
+In addition to handling customer payments, providers are responsible for paying the accumulated fees to the operator.
+
+To ensure accurate fee calculations while maintaining business confidentiality:
+
+1. Providers must construct a Zero-Knowledge Proof (ZKP) to verify the correctness of the accumulated fees based on submitted hashes.
+   These hashes should reference the cumulative bills issued to customers.
+   The ZKP should also validate that the fees align with the agreement between the provider and the operator.
+
+2. If the operator questions the reliability of the ZKP, they may request the provider to disclose the underlying financial report referenced in the ZKP.
```

Comment:
The entire point of using a ZKP would be that no underlying data must ever be disclosed. The operator just needs to know at least one control hash, which is part of the reason we publish them.

Showing the underlying data to the operator could be an interim solution if we want to implement a first version without a complex ZKP. But these two systems are very different and mutually exclusive.